azure-pipelines.yml

```diff
@@ -31,10 +31,50 @@ variables:
     value: "ukhydrographicoffice/terraform-azure-powershell-unzip:1.1.4"
   - name: DeploymentPool
     value: "UKHO Ubuntu 1804"
+  - name: coverityPool
+    value: NautilusBuild
+  - group: Covscan-vars
+
+resources:
+ repositories:
+  - repository: covscan
+    type: github
+    name: UKHO/coverityscan-buildtemplates
+    endpoint: UKHO
+    ref: refs/heads/master
 
 stages:
+  - stage: CoverityScan
+    displayName: "Coverity Scan"
+    pool:
+      name: $(coverityPool)
+    jobs:
+    - job: Coverity
+      workspace:
+        clean: all
+      steps:
+        - checkout: self
+          submodules: recursive
+        - task: UseDotNet@2
+          displayName: 'Use .NET Core sdk'
+          inputs:
+            packageType: sdk
+            version: 3.1.x
+            installationPath: $(Agent.ToolsDirectory)/dotnet
+        - checkout: covscan
+        - template: dotnet-cov.yml@covScan
+          parameters:
+            CovHostUrl: "$(CovHostUrl)"
+            CovUser: "$(CovUser)"
+            CovPwd: "$(CovPwd)"
+            StreamName: "external-notification-service"
+            BuildCommand: "$(Build.Repository.LocalPath)/external-notification-service/UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.csproj"
```

Comment:
Have you noticed if Coverity is scanning referenced projects? (like UKHO.ExternalNotificationService.Common)
If it isn't then perhaps we need a solution file here instead?

Reply:
Looking at the following has confirmed that it has scanned referenced projects:
https://sast.ukho.gov.uk:8443/reports.htm#v10946/p10054